tests/unit/services/test_lifecycle.py

```diff
@@ -808,3 +810,36 @@ def test_devel_base_no_error(fake_parts_lifecycle, build_plan_service, mocker):
 
     # Pass None as the step to ensure validation but skip the actual lifecycle run
     _ = fake_parts_lifecycle.run(None)
+
+
+class FooError(CraftError):
+    """Error to test CraftError pass through."""
+
+
+def test_override_exec(
```

Comment:
@lengau I wonder about the usefulness of this test. I wanted to showcase/test that `_exec` can be overridden but the same could be done for any other methods anyway. Do you think I should get rid of this test?